components/dropdown/dropdown-menu.js

```diff
@@ -69,6 +70,13 @@ class DropdownMenu extends ThemeMixin(DropdownContentMixin(LitElement)) {
 	}
 
 	_onFocus() {
+		if (this._useMobileStyling && !this.noMobileCloseButton && this._menuFocused) {
+			// if menu is focused and tab is clicked, move focus to the close button
+			this._menuFocused = false;
+			this.shadowRoot.querySelector('d2l-button').focus();
+			return;
+		}
+		this._menuFocused = true;
```

Comment:
I'm still a bit curious why this was necessary. Once focus moves into the menu, [it manages its own key presses](https://github.com/BrightspaceUI/core/blob/main/components/menu/menu.js#L260), but I don't see anything in there that should be suppressing the TAB key. So TAB when in the menu should move focus to the next focusable element beyond it, which _should_ be the button in this case. But maybe there's a bug in there somewhere?

Reply:
I think there is some sort of bug between the dropdown menu and the mobile tray specifically. In non-mobile mode, you can tab the button normally and exit out of the dropdown. In mobile, there is some kind of loop happening where you can only tab inside the dropdown without clicking escape. This seemed to fix it by overriding the focus to the close button... not sure what the underlying cause is for specifically the tray to have this issue ( and only with menu, trays without menus work fine).